feat(service): add cloudreve template

hassankhan2608 opened the pull request:
## Changes

- feat(service): add cloudreve template

hassankhan2608 committed:
feat(template): add Cloudreve service template

- Add Cloudreve file management and sharing system template
- Includes PostgreSQL 17 and Redis dependencies
- Proper health checks with service dependencies
- Secure password authentication without trust mode
- Logo included for UI display

ShadowArcanist commented on templates/compose/cloudreve.yaml:
```diff
@@ -0,0 +1,57 @@
+
+services:
+  cloudreve:
+    image: cloudreve/cloudreve:latest
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

```suggestion
    image: cloudreve/cloudreve:4.8.0
```

ShadowArcanist commented on templates/compose/cloudreve.yaml:
```diff
@@ -0,0 +1,57 @@
+      - CR_CONF_Redis.Server=cloudreve-redis:6379
+      - CR_CONF_Redis.Password=${SERVICE_PASSWORD_REDIS}
+    volumes:
+      - backend_data:/cloudreve/data
```
```suggestion
      - cloudreve_data:/cloudreve/data
```

ShadowArcanist commented on templates/compose/cloudreve.yaml:
```diff
@@ -0,0 +1,57 @@
+      - POSTGRES_PASSWORD=${SERVICE_PASSWORD_POSTGRESQL}
+      - POSTGRES_DB=${POSTGRES_DB:-cloudreve}
+    volumes:
+      - cloudreve-db-data:/var/lib/postgresql/data
```
```suggestion
      - cloudreve_postgres_data:/var/lib/postgresql/data
```

ShadowArcanist commented on templates/compose/cloudreve.yaml:
```diff
@@ -0,0 +1,57 @@
+    image: redis:latest
+    command: --requirepass ${SERVICE_PASSWORD_REDIS}
+    volumes:
+      - cloudreve-redis-data:/data
```
```suggestion
      - cloudreve_redis_data:/data
```

ShadowArcanist commented on templates/compose/cloudreve.yaml:
```diff
@@ -0,0 +1,57 @@
+      - backend_data:/cloudreve/data
+    healthcheck:
+      test: ["CMD", "nc", "-z", "localhost", "5212"]
+      interval: 1m30s
```
Is there any specific reason why the interval is set to too long?

```suggestion
      interval: 20s
```

ShadowArcanist reviewed (changes requested):
The app works fine, though the Compose file needs small changes for better maintainability.

hassankhan2608 committed:
Update templates/compose/cloudreve.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

hassankhan2608 committed:
Update templates/compose/cloudreve.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

hassankhan2608 committed:
Update templates/compose/cloudreve.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

hassankhan2608 committed:
Update templates/compose/cloudreve.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

hassankhan2608 committed:
Update templates/compose/cloudreve.yaml

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

peaklabs-dev committed:
Merge branch 'next' into add-cloudreve-template

peaklabs-dev committed:
feat(service): improve cloudreve

- update cloudreve image
- update postgres image
- fix formatting, order and naming

peaklabs-dev commented:
Thank you for the PR.
